cargo install php-lsp fails to compile with recent Rust (1.94+) without --locked

## Description

Running `cargo install php-lsp` fails to compile `mir-analyzer v0.1.0` with Rust 1.94.1 due to type inference errors (`E0282`) and type mismatches (`E0308`).

The root cause appears to be that without `--locked`, Cargo resolves newer versions of transitive dependencies (e.g., `php-lexer v0.3.3` instead of `v0.3.2`), which changed some types (e.g., `&str` → `Cow<'_, str>`), breaking type inference in `mir-analyzer`.

## Workaround

Using `--locked` resolves the issue:

```bash
cargo install php-lsp --locked
```

## Suggestion

It would be helpful to add `--locked` to the installation instructions in the README.md, e.g.:

```bash
cargo install php-lsp --locked
```

This ensures users get the exact dependency versions that were tested.

## Environment

- **Rust:** 1.94.1 (Arch Linux)
- **OS:** Linux 6.19.11-arch1-1
- **php-lsp:** v0.1.50

## Error Output (truncated)

```
error[E0282]: type annotations needed
  --> mir-analyzer-0.1.0/src/call.rs:36:48
   |
36 |             ExprKind::Identifier(name) => name.as_ref().to_string(),
   |                                                ^^^^^^

error[E0308]: mismatched types
   --> mir-analyzer-0.1.0/src/call.rs:183:61
    |
183 | ...entifier(name) | ExprKind::Variable(name) => name.as_ref().to_string(),
    |             ----                       ^^^^ expected `&&str`, found `&Cow<'_, str>`

error: could not compile `mir-analyzer` (lib) due to 9 previous errors
```

9 errors total across `call.rs`, `expr.rs`, `narrowing.rs`, and `stmt.rs` — all related to ambiguous `AsRef` calls and `&str` vs `Cow<'_, str>` mismatches in pattern match arms.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -41,6 +41,24 @@ jobs:
       - name: Run tests
         run: cargo test
 
+  install:
+    name: cargo install (unlocked)
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@v1
+        with:
+          toolchain: "1.94.0"
+
+      - name: cargo install without --locked
+        run: cargo install --path . --debug
+
+      - name: Verify binary
+        run: php-lsp --version
+
   lint:
     name: Lint
     runs-on: ubuntu-latest

